pkg/handlers/gitops.go

```diff
@@ -154,6 +155,18 @@ func (h *Handler) InitGitOpsConnection(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
+	// If a branch is not provided, use the default branch
+	if downstreamGitOps.Branch == "" {
+		err := gitops.UpdateDownstreamGitOps(a.ID, d.ClusterID, downstreamGitOps.RepoURI, defaultBranchName,
```

Comment:
i feel like this should be part of the "update app gitops settings" workflow instead of "test app gitops settings" workflow.

Reply:
Yes, I agree, but it would require a bigger refactor to test the repo while we perform a create or update.  Normally setting up a Gitops repo executes the following flow /Create->/Update ->/Init. We could combine them into two independent /create and /update APIs. Would like to know how @GraysonNull feels about doing this refactor on the frontend.

Reply:
Per our conversations we'll end up keeping this as a relatively low-risk change and plan to update the flow/UI later.